controller/conditional_request.go

```diff
@@ -0,0 +1,55 @@
+package controller
+
+import "time"
+
+const (
+	// IfModifiedSince the "If-Modified-Since" HTTP request header name
+	IfModifiedSince = "If-Modified-Since"
+	// LastModified the "Last-Modified" HTTP response header name
+	LastModified = "Last-Modified"
+	// IfNoneMatch the "If-None-Match" HTTP request header name
+	IfNoneMatch = "If-None-Match"
+	// ETag the "ETag" HTTP response header name
+	// should be ETag but GOA will convert it to "Etag" when setting the header.
+	// Plus, RFC 2616 specifies that header names are case insensitive:
+	// https://www.w3.org/Protocols/rfc2616/rfc2616-sec4.html#sec4.2
+	ETag = "Etag"
+	// CacheControl the "Cache-Control" HTTP response header name
+	CacheControl = "Cache-Control"
+	// MaxAge the "max-age" HTTP response header value
+	MaxAge = "max-age"
+)
+
+// ConditionalRequestContext interface with methods for conditional requests contexts
+type ConditionalRequestContext interface {
+	GenericOK(entity interface{}) error
```

Comment:
There is no reason that after the Conditional check has finished, the only available status is 200 OK.

The conditional function should only return what it knows, then delegate to a 'callback function' that triggers the next steps. The next steps could be 'Convert to REST API', Fetch additional data etc etc. It needs to hand over the return control to this function, not assume 200.